src/v/iceberg/rest_client/oauth_token.h

```diff
@@ -9,7 +9,10 @@ namespace iceberg::rest_client {
 // Oauth token returned by the catalog server, in exchange for credentials
 struct oauth_token {
```

Comment:
I don't think we need to change this just yet since presumably Polaris already supports it, but heads up that it looks like this part of the open API spec has been deprecated in favor of pinging the oauth2 server directly instead of the REST catalog. Others in the ecosystem are starting to move as well.

e.g. trino/issues/ 23086

Reply:
yes exactly, this is being deprecated, it seems this is still going to be useful for testing